### PR TITLE
First Challenge Elixir - 2nd PR

### PR DESCRIPTION
**Pull request by SebastianBurgos**
Hi, @rmonteseba , I finished the new changes with the recommendations solved, I made a new copy of the first (Challenge1.exs) in the new file (challenge_1.exs).

Here is a pant from the entire code:

code

Here is a pant from the code working:

image

### CODE REVIEW

File: challenge_1.exs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  def num_divisible_by_3_sequence(a, b) when a >= 1 and a <= b and b < 231 do
+      #Initialize the sequence in ["1"] cause the sequence always starts like that
+      sequence = obtain_sequence(["1"], 1, a, b)  #The second parameter is the variable to concat since 1
+      IO.inspect(sequence)
+      divisible_by_3_sequence = Enum.filter(sequence, fn element -> rem(String.to_integer(element), 3) == 0 end)
+      IO.puts("Number of divisibles by 3 elements in the sequence: #{length(divisible_by_3_sequence)}")
+  end
+
+  #Receive invalid parametres with a guard
+  def num_divisible_by_3_sequence(a, b) when a >= 1 or a <= b or b < 231 do
```
Comment (rmonteseba) [inline challenge_1.exs:12]: You can also delete this guard as it is pretty clear this would be the fallback case

File: challenge_1.exs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    IO.puts("Valid numbers: (1 - 231)")
+  end
+
+  #This part will receive the first part with the sequence initialized in ["1"] and the var ini in 1
```
Comment (rmonteseba) [inline challenge_1.exs:16]: ini is also on the docs

File: challenge_1.exs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  def obtain_sequence(sequence, initial_concatenator, a, b) do
+    first_element = List.last(obtain_first_sequence(sequence, initial_concatenator, a)) #Get the first element of the final sequence (the last from the first sequence)
+    construct_sequence([first_element], a, b) #Construct the final sequence since the first element
+  end
+
+  #Returns the first sequence when the sumator == a
+  def obtain_first_sequence(sequence, a, a) do sequence end
+
+  def obtain_first_sequence(sequence, sumator, a) do
```
Comment (rmonteseba) [inline challenge_1.exs:25]: This method and construct_sequence are actually doing the same, is there someway we can use only on of them?

File: challenge_1.exs (hunk 1 above)
Comment (SebastianBurgos) [reply challenge_1.exs:12]: code

File: challenge_1.exs (hunk 3 above)
Comment (SebastianBurgos) [reply challenge_1.exs:25]: Solved

**Pull request closed without merging**